src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp

```diff
@@ -53,61 +52,130 @@ TypeLookupReplyListener::~TypeLookupReplyListener()
 {
 }
 
+void TypeLookupReplyListener::check_get_types_reply(
+        const SampleIdentity& request_id,
+        const TypeLookup_getTypes_Out& reply)
+{
+    // Check if the received reply SampleIdentity corresponds to an outstanding request
+    auto requests_it = typelookup_manager_->async_get_type_requests_.find(request_id);
+    if (requests_it != typelookup_manager_->async_get_type_requests_.end())
+    {
+        for (xtypes::TypeIdentifierTypeObjectPair pair : reply.types())
+        {
+            if (RETCODE_OK == fastrtps::rtps::RTPSDomainImpl::get_instance()->type_object_registry_observer().
+                            register_type_object(pair.type_identifier(), pair.type_object()))
+            {
+                // Notify the callbacks associated with the request
+                typelookup_manager_->notify_callbacks(requests_it->second);
+            }
```

Comment:
What happens if the registering fails? Could it fail? Failures would be if the TypeIdentifier/TypeObject discriminator differs or if the TypeIdentifier does not correspond to the given TypeObject. Even if it is really weird if this occurs, we should check and communicate the error to the user in some way.

Reply:
added error check in [29681bc](https://github.com/eProsima/Fast-DDS/pull/4240/commits/29681bc3ecb5a0cf1f4e298b7bf093bff1ab1753)